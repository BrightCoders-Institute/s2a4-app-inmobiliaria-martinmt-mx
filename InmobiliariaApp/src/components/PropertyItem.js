import React from 'react';
import { useState } from 'react';
import { View, Text, StyleSheet, TouchableOpacity, Image } from 'react-native';
import { propertyItemStyles } from './PropertyItemStyles'
import PropTypes from 'prop-types';

const PropertyItem = ({ property, onPress }) => {
    const [liked, setLiked] = useState(false);

    const handleLike = () => {
      setLiked(!liked);
    };

    return (
      <TouchableOpacity onPress={onPress} style={propertyItemStyles.card}>
        <View style={propertyItemStyles.imageContainer}>
            <Image source={{ uri: property.image }} style={propertyItemStyles.image} resizeMode="cover" />
            <View style={propertyItemStyles.ratingContainer}>
                <Image source={require('../../assets/icons/star.png')} style={propertyItemStyles.starIcon} />
                <Text style={propertyItemStyles.ratingText}>{property.rating}</Text>
            </View>
        </View>
        <View style={propertyItemStyles.detailsContainer}>
          <Text style={propertyItemStyles.title}>{property.name}</Text>
          <View style={propertyItemStyles.addressContainer}>
            <Image source={require('../../assets/icons/LocationIcon.png')} style={propertyItemStyles.locationIcon} />
            <Text style={propertyItemStyles.addressText}>{property.address}</Text>
          </View>
          <View style={propertyItemStyles.infoContainer}>
            <View style={propertyItemStyles.iconContainer}>
              <Image source={require('../../assets/icons/RoomIcon.png')} style={propertyItemStyles.icon} />
              <Text style={propertyItemStyles.infoText}> {property.bedrooms}</Text>
            </View>
            <View style={propertyItemStyles.iconContainer}>
              <Image source={require('../../assets/icons/BathIcon.png')} style={propertyItemStyles.icon} />
              <Text style={propertyItemStyles.infoText}> {property.bathrooms}</Text>
            </View>
            <View style={propertyItemStyles.iconContainer}>
              <Image source={require('../../assets/icons/AreaIcon.png')} style={propertyItemStyles.icon} />
              <Text style={propertyItemStyles.infoText}> {property.area} ftÂ²</Text>
            </View>
          </View>
          <Text>{property.rent}/m</Text>
          <View style={propertyItemStyles.likeContainer}>
          <TouchableOpacity onPress={handleLike}>
            <Image
              source={liked ? require('../../assets/icons/like_filled.png') : require('../../assets/icons/like.png')}
              style={propertyItemStyles.likeIcon}
            />
          </TouchableOpacity>
        </View>
        </View>
      </TouchableOpacity>
    );
  };

const styles = StyleSheet.create({
  
});

PropertyItem.propTypes = {
  property: PropTypes.shape({
    image: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    address: PropTypes.string.isRequired,
    bedrooms: PropTypes.number.isRequired,
    bathrooms: PropTypes.number.isRequired,
    area: PropTypes.string.isRequired,
    rent: PropTypes.string.isRequired,
  }).isRequired,
  onPress: PropTypes.func.isRequired,
};

export default PropertyItem;
