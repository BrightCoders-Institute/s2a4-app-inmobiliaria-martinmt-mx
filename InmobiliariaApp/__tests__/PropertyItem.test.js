import React from 'react';
import {render} from '@testing-library/react-native'
import PropertyItem from '../src/components/PropertyItem';

describe('PropertyItem', () => {
  it('renders correctly', () => {
    const property = {
      id: 1,
      name: 'Casa de Prueba',
      address: 'Calle de Prueba #123',
      bedrooms: 3,
      bathrooms: 2,
      area: '200',
      rent: '$1000',
      image: 'https://example.com/image1.jpg',
      rating: 4,
    };

    const { toJSON } = render(<PropertyItem property={property} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
