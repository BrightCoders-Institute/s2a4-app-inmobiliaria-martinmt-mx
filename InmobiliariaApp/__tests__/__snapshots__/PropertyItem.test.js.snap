// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyItem renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgb(245, 253, 255)",
      "borderColor": "#ccc",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "margin": 5,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "borderRadius": 8,
        "height": 100,
        "marginRight": 10,
        "overflow": "hidden",
        "position": "relative",
        "width": 100,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        {
          "uri": "https://example.com/image1.jpg",
        }
      }
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(251, 237, 183, 1)",
          "borderRadius": 15,
          "bottom": 10,
          "flexDirection": "row",
          "left": "44%",
          "paddingHorizontal": 7,
          "paddingVertical": 5,
          "position": "absolute",
          "transform": [
            {
              "translateX": -15,
            },
          ],
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/star.png",
          }
        }
        style={
          {
            "height": 15,
            "marginRight": 3,
            "tintColor": "rgba(238, 186, 0, 1)",
            "width": 15,
          }
        }
      />
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        4
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    >
      Casa de Prueba
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/LocationIcon.png",
          }
        }
        style={
          {
            "height": 15,
            "marginRight": 5,
            "width": 15,
          }
        }
      />
      <Text
        style={
          {
            "fontSize": 16,
          }
        }
      >
        Calle de Prueba #123
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginVertical": 5,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginRight": 15,
            "marginVertical": 5,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/RoomIcon.png",
            }
          }
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
        <Text
          style={
            {
              "marginLeft": 5,
            }
          }
        >
           
          3
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginRight": 15,
            "marginVertical": 5,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/BathIcon.png",
            }
          }
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
        <Text
          style={
            {
              "marginLeft": 5,
            }
          }
        >
           
          2
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginRight": 15,
            "marginVertical": 5,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/AreaIcon.png",
            }
          }
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
        <Text
          style={
            {
              "marginLeft": 5,
            }
          }
        >
           
          200
           ftÂ²
        </Text>
      </View>
    </View>
    <Text>
      $1000
      /m
    </Text>
    <View
      style={
        {
          "bottom": 10,
          "position": "absolute",
          "right": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/like.png",
            }
          }
          style={
            {
              "height": 20,
              "tintColor": "green",
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
